### ISSUE TITLE
How to auth Prometheus in Thanos

### ISSUE THREAD

**Issue by shineshane** [thanos-io/thanos#4671]
My Prometheus instance has enable basic-auth: https://prometheus.io/docs/guides/basic-auth/

when I run Thanos sidecar:
```
thanos sidecar --tsdb.path /home/prometheus/data/ --prometheus.url "http://localhost:9090"
```

get log:
```
level=warn ts=2021-09-17T08:56:33.579018157Z caller=sidecar.go:159 msg="failed to fetch prometheus version. Is Prometheus running? Retrying" err="expected 2xx response, got 401. Body: Unauthorized\n"
```

I looked up the flags in `thanos sidecar -h` but the flag related to prometheus auth could not be found. How should I configure Thanos?

**Comment by Namanl2001** [comment]
This feature is not present in thanos yet, however we have a WIP PR for it.
Related issue: https://github.com/thanos-io/thanos/issues/3975 
WIP PR: https://github.com/thanos-io/thanos/pull/4612

**Comment by yeya24** [comment]
Closed by #4612
